Src/VsVimShared/Implementation/IntelliCode/IntelliCodeCommandTarget.cs

```diff
@@ -13,44 +13,93 @@
-                {
-                    _lookedForCascadingCompletionSource = true;
-                    _cascadingCompletionSource = TextView
-                        .Properties
-                        .PropertyList
-                        .Where(pair => pair.Key is Type { Name: "CascadingCompletionSource" })
-                        .Select(x => x.Value)
-                        .FirstOrDefault();
-                }
-
-                return _cascadingCompletionSource;
+                EnsureLookedForData();
+                return _data?.CascadingCompletionSource;
             }
         }
 
         internal IntelliCodeCommandTarget(
             IVimBufferCoordinator vimBufferCoordinator,
-            IAsyncCompletionBroker asyncCompletionBroker)
+            IAsyncCompletionBroker asyncCompletionBroker,
+            ITextDocumentFactoryService textDocumentFactoryService)
         {
             VimBufferCoordinator = vimBufferCoordinator;
             AsyncCompletionBroker = asyncCompletionBroker;
+            TextDocumentFactoryService = textDocumentFactoryService;
+        }
+
+        private void EnsureLookedForData()
+        {
+            if (_lookedForData)
+            {
+                return;
+            }
+
+            _lookedForData = true;
+            try
+            {
+                var cascadingCompletionSource = TextView
+                    .Properties
+                    .PropertyList
+                    .Where(pair => pair.Key is Type { Name: "CascadingCompletionSource" })
+                    .Select(x => x.Value)
+                    .FirstOrDefault();
+                if (cascadingCompletionSource is object)
+                {
+                    var type = cascadingCompletionSource.GetType();
+                    var currentPrediction = type.GetField("_currentPrediction", BindingFlags.NonPublic | BindingFlags.Instance);
+                    var completionState = type.GetField("_completionState", BindingFlags.NonPublic | BindingFlags.Instance);
+                    if (currentPrediction is object && completionState is object)
+                    {
+                        _data = (cascadingCompletionSource, currentPrediction, completionState);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                VimTrace.TraceInfo($"IntelliCodeCommandTarget::EnsureLookedForData error {ex.Message}");
+                Debug.Assert(false);
+            }
+        }
+
+        private bool IsIntelliCodeHandlingEscape()
+        {
+            EnsureLookedForData();
+            if (_data is { } data)
+            {
+                try
+                {
+                    var completionState = data.CompletionState.GetValue(data.CascadingCompletionSource);
+                    var currentPrediction = data.CurrentPrediction.GetValue(data.CascadingCompletionSource);
+                    return completionState is object || currentPrediction is object;

```

Comment:
```suggestion
                    return completionState is not null || currentPrediction is not null;

```
Someone should write a blog post titled "`is object` considered harmful"

Reply:
Nah, someone already tweeted that it's the best option 😄